node/src/components/consensus/highway_core/evidence.rs

```diff
@@ -47,32 +77,78 @@ impl<C: Context> Evidence<C> {
     /// the two units are the same.
     pub(crate) fn validate(
         &self,
-        v_id: &C::ValidatorId,
+        validators: &Validators<C::ValidatorId>,
         instance_id: &C::InstanceId,
     ) -> Result<(), EvidenceError> {
         match self {
             Evidence::Equivocation(unit1, unit2) => {
-                if unit1.wire_unit.creator != unit2.wire_unit.creator {
-                    return Err(EvidenceError::EquivocationDifferentCreators);
+                Self::validate_equivocation(unit1, unit2, instance_id, validators)
+            }
+            Evidence::Endorsements {
+                endorsement1,
+                unit1,
+                endorsement2,
+                unit2,
+                swimlane2,
```

Comment:
Do we have a task to limit the acceptable length of the swimlane?

Reply:
I created [HWY-212](https://casperlabs.atlassian.net/browse/HWY-212).